src/core/plugins/auth/actions.js

```diff
@@ -83,15 +83,19 @@ export const authorizePassword = ( auth ) => ( { authActions } ) => {
   } else {
     Object.assign(form, {username}, {password})
 
-    if ( passwordType === "query") {
-      if ( clientId ) {
-        query.client_id = clientId
+    if ( clientId && clientSecret ) {
```

Comment:
This logic could break some of the previous cases...
We had separate conditions for clientId & clientSecret, not it forces both

Reply:
Is it a valid use case to only supply one of those values, though?

I ask only because the branch of code that was previously defaulting to putting these in the `Authorization` header _didn't_ include similar checks.

So in the case where where type was `request-body`, and only a `client_id` value was supplied, it previously would have set the header to `Basic " + btoa("the_client_id:undefined")`, which sounds like another lurking bug.

Happy to add back in the separate conditions, but I wanted to double check that supplying one or other value (id or secret, but not both) is something we're expecting people could be doing on purpose?